src/MongoDB/Manager.c

```diff
@@ -289,7 +289,11 @@ static bool php_phongo_manager_select_server(bool for_writes, zval* zreadPrefere
 	}
 
 	if (!for_writes) {
-		read_preference = zreadPreference ? phongo_read_preference_from_zval(zreadPreference TSRMLS_CC) : mongoc_client_get_read_prefs(client);
+		if (zreadPreference) {
+			read_preference = phongo_read_preference_from_zval(zreadPreference TSRMLS_CC);
+		} else if (inherit_read_preference) {
+			read_preference = mongoc_client_get_read_prefs(client);
+		}
 	}
 
 	selected_server = mongoc_client_select_server(client, for_writes, read_preference, &error);
```

Comment:
According to the [`mongoc_client_select_server` docs](http://mongoc.org/libmongoc/current/mongoc_client_select_server.html), the client's read preference is used if `read_preference` is `NULL`. That said, I traced the function calls through to `mongoc_topology_description_suitable_servers`:

 * mongoc_client_select_server
 * mongoc_topology_select
 * mongoc_topology_select_server_id
 * mongoc_topology_description_select
 * mongoc_topology_description_suitable_servers

Therein, we have:

```c
mongoc_read_mode_t read_mode = mongoc_read_prefs_get_mode (read_pref);
```

If `read_pref` is `NULL` here, a primary RP is used. This would seem to conflict with the `mongoc_client_select_server` docs, unless I'm missing something. Maybe @kevinAlbs can chime in.

Either libmongoc has a bug in its implementation or the docs need to be corrected, but I think the safest thing to do in PHPC is to default to a primary read preference explicitly.

Reply:
I agree with your assessment. `mongoc_client_select_server` does not inherit the client's  read preference, but defaults to primary read preference. The documentation does not match the implementation. I created https://jira.mongodb.org/browse/CDRIVER-3732 to update the documentation and validate with a test.

Reply:
@alcaeus I'll defer to you if you want to hard-code a primary RP here or leave the code as-is. Either way, I'm confident that the included test ensures correctness.